src/vs/platform/windows/electron-main/windowImpl.ts

```diff
@@ -139,7 +139,7 @@ export class CodeWindow extends Disposable implements ICodeWindow {
 			return profile;
 		}
 
-		return this.userDataProfilesService.getOrSetProfileForWorkspace(this.config.workspace ?? 'empty-window', this.userDataProfilesService.defaultProfile);
+		return this.userDataProfilesService.getProfileForWorkspace(this.config.workspace ?? toWorkspaceIdentifier(this.backupPath, this.isExtensionDevelopmentHost)) ?? this.userDataProfilesService.defaultProfile;
```

Comment:
I actually was recently in this area to investigate if we could have a `IAnyWorkspaceIdentifier` on the window which includes the empty workspace case. Having to create the workspace identifier here when in empty case is quite ugly, so maybe I should revisit that again...

Reply:
Yeah, that was initial approach to use `IAnyWorkspaceIdentifier` for `INativeWindowConfiguration.worksapce` and `ICodeWindow.openedWorkspace`. But it was more tricky and non-trivial for me to do that with this change set. It would be great if you can do that or I can create another PR for that.

Reply:
I can look into it but not sure it will happen, I remember some roadblocks on the way.

Reply:
Yeah it has a ripple that I am a bit too afraid to push given possible regressions, so will keep it as is.